tensorflow_datasets/core/features/image_feature.py

```diff
@@ -146,6 +158,9 @@ def encode_example(self, image_or_path_or_fobj):
 
   def decode_example(self, example):
     """Reconstruct the image from the tf example."""
+    if self._encoding_format is 'png' and self._dtype is tf.uint16:
```

Comment:
You should not use `is` when you want to test equality. The fact that it sometimes works due to the fact that Python is caching string make this behavior unpredictable.

See: https://stackoverflow.com/questions/1504717/why-does-comparing-strings-using-either-or-is-sometimes-produce-a-differe